src/predefined-variables.ts

```diff
@@ -13,16 +13,29 @@ export function init ({gitData, argv, envMatchedVariables}: PredefinedVariablesO
     // 1. gitlab variables files
     // 2. values derieved implicitly from `git remote -v`
     // 3. default value
+    const CI_SERVER_PROTOCOL = envMatchedVariables["CI_SERVER_PROTOCOL"]
+      ?? ((gitData.remote.schema === "http" || gitData.remote.schema === "https")
+        ? gitData.remote.schema
+        : "https");
     const CI_SERVER_PORT = envMatchedVariables["CI_SERVER_PORT"]
-      ?? (gitData.remote.schema === "http" || gitData.remote.schema === "https")
+      ?? ((gitData.remote.schema === "http" || gitData.remote.schema === "https")
         ? gitData.remote.port
-        : "443";
+        : "443");
     const CI_SERVER_SHELL_SSH_PORT = envMatchedVariables["CI_SERVER_SHELL_SSH_PORT"]
-      ?? (gitData.remote.schema === "ssh")
+      ?? ((gitData.remote.schema === "ssh")
         ? gitData.remote.port
-        : "22";
+        : "22");
+    const CI_SERVER_HOST = envMatchedVariables["CI_SERVER_HOST"]
+      ?? `${gitData.remote.host}`;
+    const CI_SERVER_FQDN = envMatchedVariables["CI_SERVER_FQDN"]
+      ?? `${CI_SERVER_HOST}:${CI_SERVER_PORT}`;
```

Comment:
```suggestion
    const CI_SERVER_FQDN = envMatchedVariables["CI_SERVER_FQDN"]
      ?? (CI_SERVER_PORT == "443"
          ? gitData.remote.host
          : `${gitData.remote.host}:${CI_SERVER_PORT}`);
```
to be consistent with gitlab.com... the port for the FQDN is implicit when it's 443 

as seen https://github.com/firecow/gitlab-ci-local/pull/1368#issuecomment-2398753437